drivers/telescope/lx200classic.cpp

```diff
@@ -544,11 +500,68 @@ bool LX200Classic::ReadScopeStatus()
     { 
         if ((TrackState == SCOPE_PARKED) && (curTrackState == SCOPE_PARKING) && !isSimulation())
         {
-            setAlignmentMode(PortFD, LX200_ALIGN_LAND);
+            if (setAlignmentMode(PortFD, LX200_ALIGN_LAND) < 0)
+            {
+                LOG_ERROR("Parking Failed.");
+                AlignmentSP.s = IPS_ALERT;
+                IDSetSwitch(&AlignmentSP, "Error setting alignment mode.");
+                return false;
+            }
             //Update the UI
             LX200Generic::getAlignment();
         }
     }
 
     return true;
 }
+
+void LX200Classic::azAltToRaDecNow(double az, double alt, double &ra, double &dec)
```

Comment:
This! Yes! 🖖😌

Reply:
If you like to reuse, can you figure out where to move to?

Reply:
Yep, let me find a good spot for it... (will probably find it already exists :D )

Reply:
TransformCelestialToTelescope() does this, looks like it is provided by alignment/AlignmentSubsystemForDrivers.h (DSC, Temma, and Skywatcher are the only folks that are using it, I will give it a try)

Reply:
It looks like LX200Telescope just calls "slewToPark()" (a direct LX200 command), and several of the child classes don't support that, and had to implement Park() and UnPark() from scratch. 
Given the way the classes are laid out, LX200Generic is the only place in between the two, but that seems to want to be an empty passthrough class.

This behavior is implemented in indicom.h and there's a really gorgeous implementation in alignment/*Math*.h, and all the ln_ decorators are because these are functions directly from libnova.

Maybe there should be LX200Telescope::SlewToPark() which either calls slewToPark() (the internal command), or this snippet.

Or maybe this is LX200Generic::Park() -- the class should stay empty, but this is a bit that's not in the base that is in several of the children (Classic, FS-2, SS2000PC, AP, AP_GTOCP2, AP_Experimental). 

I don't know the right answer -- just seems like if the answer going forward for parking will be "copy paste these exact 5 functions", we've built the object model wrong somehow...

Reply:
I went for two days back and forth through the code stack to figure it out. With the classic there were three aspects different form assumption made elsewhere:

- The slew to park command as you mention
- Lack of setting positions in AzAlt. Only RaDec possible
- The mount can be AltAz or Polar, and therefore needs to remember the state.

The LX200telescope class may be able to implement (part of) this. However it seems the situation is that there are a lot of different choices made by mount manufacturers.

Reply:
Agreed, I wonder about “intent” in the LX200 family. I feel like LX200Telescope is a bit prescriptive, but that’s  what you get when everyone copies an existing standard... sort of. Sigh. Maybe this just gets copy pasta

Reply:
I have proposed to move the AltAz<->Equatorial conversion to another location, but with another pull request. Lets take time to figure out the best location.